.github/actions/start-opensearch/action.yml

```diff
@@ -0,0 +1,175 @@
+        chmod +x plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        yes | plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+        chown 1001:1001 -R /opensearch
+        su -c "/opensearch/bin/opensearch" -s /bin/bash opensearch
+        EOF
+        docker build -t opensearch-test:latest -f- . <<EOF
+        FROM ubuntu:latest
+        COPY --chown=1001:1001 os-ep.sh /docker-host/
+        COPY --chown=1001:1001 ${{ inputs.plugin-name }}.zip /docker-host/${{ inputs.plugin-zip }}.zip
+        COPY --chown=1001:1001 opensearch* /opensearch/
+        RUN chmod +x /docker-host/os-ep.sh
+        RUN useradd -u 1001 -s /sbin/nologin opensearch
+        ENV PATH="/opensearch/bin:${PATH}"
+        WORKDIR /opensearch/
+        ENTRYPOINT /docker-host/os-ep.sh
+        EOF
+      shell: bash
+
+    - name: Install Plugin into OpenSearch for Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: |
+        'y' | .\opensearch\bin\opensearch-plugin.bat install file:\a\${{ github.event.repository.name }}\${{ github.event.repository.name }}\${{ inputs.plugin-name }}.zip
+        'y', 'y', 'N' | .\opensearch\plugins\${{ inputs.plugin-name }}\tools\${{ inputs.plugin-start-script }}.bat
+      shell: pwsh
+
+    - name: Install Plugin on MacOS
+      if: ${{ inputs.platform == 'macos-latest' }}
+      run: |
+        chmod +x ./opensearch/bin/opensearch-plugin
+        yes | ./opensearch/bin/opensearch-plugin install file:/a/{{ github.event.repository.name }}/{{ github.event.repository.name }}/${{ inputs.plugin-name }}.zip
+        chmod +x ./opensearch/plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh   
+        yes yes No | ./opensearch/plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
+      shell: bash
+
+    # Run OpenSearch
+    - name: Run OpenSearch with plugin on Linux
+      if: ${{ inputs.platform == 'ubuntu-latest'}}
+      run: docker run --name ops -d -p 9200:9200 -p 9600:9600 -i opensearch-test:latest
+      shell: bash
+
+    - name: Run OpenSearch with plugin on Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: start .\opensearch\bin\opensearch.bat
+      shell: pwsh
+
+    - name: Run OpenSearch with plugin on MacOS
+      if: ${{ inputs.platform == 'macos-latest'}}
+      run: |
+        chmod +x ./Opensearch/bin/opensearch
+        ./Opensearch/bin/opensearch
+      shell: bash
+
+    # Give the OpenSearch process some time to boot up before sending any requires, might need to increase the default time!
+    - name: Sleep while OpenSearch finishes starting up
+      if: ${{ inputs.platform != 'windows-latest'}}
+      uses: whatnick/wait-action@v0.1.2
+      with:
+        time: '30s'
+
+    - name: Sleep while OpenSearch starts on Windows
+      if: ${{ inputs.platform == 'windows-latest'}}
+      run: Start-Sleep -s 30
+      shell: pwsh
```

Comment:
Doesn't `whatnick/wait-action@v0.1.2` work on all platforms so we can condense this into one step?

Reply:
It does not, it only works on Linux.

Reply:
Nit; If you are feeling adventurous https://github.com/peternied/action-sleep

Reply:
This seems like the code that was already in the file but I will put the action you just made in the spot if you think that is best.